timelock/program/src/instruction.rs

```diff
@@ -0,0 +1,590 @@
+    ///   8. `[writable]` Voting mint account.
+    ///   9. `[writable]` Yes Voting mint account.
+    ///   10. `[writable]` No Voting mint account.
+    ///   11. `[]` Timelock state account.
+    ///   12. `[]` Timelock set account.
+    ///   13. `[]` Transfer authority
+    ///   14. `[]` Yes Transfer authority
+    ///   15. `[]` No Transfer authority
+    ///   16. `[]` Timelock program mint authority
+    ///   17. `[]` Timelock program account pub key.
+    ///   18. `[]` Token program account.
+    WithdrawVotingTokens {
+        /// How many voting tokens to withdrawal
+        voting_token_amount: u64,
+    },
+
+    ///   0. `[writable]` Timelock config key. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, governance mint key, council mint key, timelock program account key.
+    ///   1. `[]` Program account that this config uses
+    ///   2. `[]` Governance mint that this config uses
+    ///   3. `[]` Council mint that this config uses [Optional] [Pass in 0s otherwise]
+    ///   4. `[]` Timelock program account pub key.
+    ///   5. `[]` Token program account.
+    InitTimelockConfig {
+        /// Consensus Algorithm
+        consensus_algorithm: u8,
+        /// Execution type
+        execution_type: u8,
+        /// Timelock Type
+        timelock_type: u8,
+        /// Voting entry rule
+        voting_entry_rule: u8,
+        /// Minimum slot time-distance from creation of proposal for an instruction to be placed
+        minimum_slot_waiting_period: u64,
+        /// Time limit in slots for proposal to be open to voting
+        time_limit: u64,
+        /// Optional name
+        name: [u8; CONFIG_NAME_LENGTH],
+    },
+
+    ///   0. `[]` Timelock config key. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, governance mint key, council mint key, and timelock program account key.
+    ///   1. `[]` Program account to tie this config to.
+    ///   2. `[]` Governance mint to tie this config to
+    ///   3. `[]` Council mint [optional] to tie this config to [Pass in 0s otherwise]
+    ///   4. `[]` Payer
+    ///   5. `[]` Timelock program account pub key.
+    ///   6. `[]` Timelock program pub key. Different from program account - is the actual id of the executable.
+    ///   7. `[]` Token program account.
+    ///   8. `[]` System account.
+    CreateEmptyTimelockConfig,
+
+    ///   0. `[]` Governance voting record key. Needs to be set with pubkey set to PDA with seeds of the
+    ///           program account key, proposal key, your voting account key.
+    ///   1. `[]` Proposal key
+    ///   2. `[]` Your voting account
+    ///   3. `[]` Payer
+    ///   4. `[]` Timelock program account pub key.
+    ///   5. `[]` Timelock program pub key. Different from program account - is the actual id of the executable.
```

Comment:
This is passed into the instruction naturally as the first parameter, is the actual executable of the program needed?

Reply:
Yeah but that's a Pubkey, not an AccountINfo. We need the AccountInfo here for the logic.

Reply:
What do you need it for? I looked at the call to `create_account_raw` and that doesn't need the calling program

Reply:
I'd have to look. It may be that this is a vestigial requirement from code that was later removed. Thanks for pointing it out, I'll get back to you on it once Seb rolls through these review comments after all the major refactoring on my list.

Reply:
Governance program account was removed from inout accounts